Don't emit AArch64 rbit when hipcc is compiling AMDGPU fatbins.
hipcc on an ARM host still defines __aarch64__ while targeting gfx*.

--- a/intern/cycles/util/math_base.h
+++ b/intern/cycles/util/math_base.h
@@ -817,11 +817,12 @@
   return __brev(x);
 #elif defined(__KERNEL_METAL__)
   return reverse_bits(x);
-#elif defined(__aarch64__) || (defined(_M_ARM64) && !defined(_MSC_VER))
+#elif !defined(__HIPCC__) && (defined(__aarch64__) || (defined(_M_ARM64) && !defined(_MSC_VER)))
   /* Assume the rbit is always available on 64bit ARM architecture. */
   __asm__("rbit %w0, %w1" : "=r"(x) : "r"(x));
   return x;
-#elif defined(__arm__) && ((__ARM_ARCH > 7) || __ARM_ARCH == 6 && __ARM_ARCH_ISA_THUMB >= 2)
+#elif !defined(__HIPCC__) && defined(__arm__) && \
+    ((__ARM_ARCH > 7) || __ARM_ARCH == 6 && __ARM_ARCH_ISA_THUMB >= 2)
   /* This ARM instruction is available in ARMv6T2 and above.
    * This 32-bit Thumb instruction is available in ARMv6T2 and above. */
   __asm__("rbit %0, %1" : "=r"(x) : "r"(x));
